pkg/dclient/client.go

```diff
@@ -266,6 +273,25 @@ type ServerPreferredResources struct {
 	cachedClient discovery.CachedDiscoveryInterface
 }
 
+//Poll will keep invalidate the local cache
+func (c ServerPreferredResources) Poll(resync time.Duration, stopCh <-chan struct{}) {
+	// start a ticker
+	ticker := time.NewTicker(resync)
+	defer func() { ticker.Stop() }()
+	glog.Infof("Starting registered resources sync: every %d seconds", resync)
+	for {
+		select {
+		case <-stopCh:
+			glog.Info("Stopping registered resources sync")
+			return
+		case <-ticker.C:
+			// set cache as stale
+			glog.V(6).Info("invalidating local client cache for registered resources")
+			c.cachedClient.Invalidate()
```

Comment:
Got a question, where do we update the cache if we clear it every 10 seconds?

Reply:
Every call done via dynamic client will first refer local cache for registered resources and if not found then the query api-server to refresh local cache.